packages/expression-evaluator/lib/evaluators/ExpressionEvaluatorFactory.ts

```diff
@@ -1,42 +1,91 @@
+    const res: FunctionExpression | undefined = {
+      ...regularFunctions,
+      ...specialFunctions,
+      ...namedFunctions,
+    }[<C.NamedOperator | C.Operator> functionName];
+    if (res) {
+      return res;
+    }
+    const extensionFinder: ((functionNamedNode: RDF.NamedNode) =>
+    Promise<((args: RDF.Term[]) => Promise<RDF.Term>) | undefined>) | undefined =
+      context.get(KeysInitQuery.extensionFunctionCreator);
+    if (extensionFinder) {
+      const definition = await extensionFinder(new DataFactory<RDF.Quad>().namedNode(functionName));
+      if (definition) {
+        return new NamedExtension(functionName, definition);
+      }
+    }
+    const extensionMap: Record<string, (args: RDF.Term[]) => Promise<RDF.Term>> | undefined =
+      context.get(KeysInitQuery.extensionFunctions);
+    if (extensionMap) {
+      const definition = extensionMap[functionName];
+      if (definition) {
+        return new NamedExtension(functionName, definition);
+      }
+    }
+    throw new Error('No Function Actor Replied');
+  };
+
+  public readonly orderByBus: OrderByBus = async({ context, getSuperType }) =>
+    new OrderByEvaluator(new ContextualizedEvaluator({
+      now: context.get(KeysInitQuery.queryTimestamp),
+      baseIRI: context.get(KeysInitQuery.baseIRI),
+      functionArgumentsCache: context.get(KeysInitQuery.functionArgumentsCache),
+      actionContext: context,
+      mediatorQueryOperation: this.mediatorQueryOperation,
+      mediatorFunction: this.functionsBus,
+      getSuperType,
+    }),
+    <ITermFunction> await this.functionsBus({ functionName: RegularOperator.EQUAL, context, definitionType: 'onTerm' }),
+    <ITermFunction> await this.functionsBus({ functionName: RegularOperator.LT, context, definitionType: 'onTerm' }));
 
   public constructor(args: IExpressionEvaluatorFactoryArgs) {
     this.mediatorBindingsAggregatorFactory = args.mediatorBindingsAggregatorFactory;
     this.mediatorQueryOperation = args.mediatorQueryOperation;
   }
 
   // TODO: remove legacyContext in *final* update (probably when preparing the EE for function bussification)
-  public createEvaluator(algExpr: Alg.Expression, context: IActionContext,
-    legacyContext: Partial<IAsyncEvaluatorContext> = {}): ExpressionEvaluator {
-    return new ExpressionEvaluator(algExpr, {
-      now: context.get(KeysInitQuery.queryTimestamp),
-      baseIRI: context.get(KeysInitQuery.baseIRI),
-      functionArgumentsCache: context.get(KeysInitQuery.functionArgumentsCache),
-      actionContext: context,
-      bnode: (input?: string) => Promise.resolve(new BlankNodeBindingsScoped(input || `BNODE_${bnodeCounter++}`)),
-      exists: ActorQueryOperation.createExistenceResolver(context, this.mediatorQueryOperation),
-      ...legacyContext,
-    });
+  public async createEvaluator(algExpr: Alg.Expression, context: IActionContext,
+    contextEval?: ContextualizedEvaluator,
```

Comment:
Can/should we enforce `contextEval` to be defined?
So we can avoid cases where we are continuously recreating `ContextualizedEvaluator` due to an implementation oversight?

Reply:
I don't know why I had that argument anymore. It wasn't used. It's not expensive to recreate either.